tests/kernel/update_kernel.pm

```diff
@@ -166,74 +162,70 @@ sub install_lock_kernel {
         $package =~ s/$/-$l_v/;
     }
 
-    # workaround for SLE15 - 4.12.4-25.13.1
-    if ($wk_ker == 1) {
-        @packages = grep { $_ ne 'kernel-source-4.12.14-25.13.1' } @packages;
-    }
-
     # install and lock needed kernel
     zypper_call("in " . join(' ', @packages), exitcode => [0, 102, 103, 104], timeout => 1400);
     zypper_call("al " . join(' ', @lpackages));
 }
 
 sub prepare_kgraft {
     my ($repo, $incident_id) = @_;
-    my $arch    = get_required_var('ARCH');
-    my $version = get_required_var('VERSION');
-    my $release_override;
-    my $lp_product;
-    my $lp_module;
-    if ($version eq '12') {
-        $release_override = '-d';
-    }
-    if (!is_sle('>=12-SP3')) {
-        $version = '12';
-    }
-    # SLE15 has different structure of modules and products than SLE12
-    if (is_sle('15+')) {
-        $lp_product = 'sle-module-live-patching';
-        $lp_module  = 'SLE-Module-Live-Patching';
-    }
-    else {
-        $lp_product = 'sle-live-patching';
-        $lp_module  = 'SLE-Live-Patching';
-    }
 
-    #install kgraft product
-    zypper_call("ar http://download.suse.de/ibs/SUSE/Products/$lp_module/$version/$arch/product/ kgraft-pool");
-    zypper_call("ar $release_override http://download.suse.de/ibs/SUSE/Updates/$lp_module/$version/$arch/update/ kgraft-update");
-    zypper_call("ref");
-    zypper_call("in -l -t product $lp_product", exitcode => [0, 102, 103]);
-    zypper_call("mr -e kgraft-update");
+    fully_patch_system;
```

Comment:
Doing full update here is too early. Validate the incident repos first.

Reply:
Can you clarify what you mean by "validating the incident repos"? Searching them for the incident KLP package?

Reply:
Simply move `fully_patch_system;` immediately before `install_lock_kernel();` as it is currently in the master branch. System update can take around 30 minutes so it's a waste of time and resources if the test will fail either way because the incident repo is empty.

Reply:
Ok, understood.

Reply:
Done.